### PR TITLE
Ibp 4440 move attributes to ontology

### PR DESCRIPTION
**Pull request by clarysabel**

### CODE REVIEW

File: src/main/java/org/generationcp/commons/ruleengine/naming/expression/AttributeSourceExpression.java [hunk 1]
```diff
@@ -17,18 +16,17 @@
 
 	public static final String ATTRIBUTE_KEY = "ATTRSC";
 	public static final String PATTERN_KEY = "\\[" + ATTRIBUTE_KEY + "\\.([^\\.]*)\\]"; // Example: ATTRSC.NOTES
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/commons/ruleengine/naming/expression/AttributeSourceExpression.java:18]: Modify the example to `ATTRSC.<integer>`

File: src/main/java/org/generationcp/commons/workbook/generator/CodesSheetGenerator.java [hunk 2]
```diff
@@ -15,6 +13,8 @@
 import org.generationcp.middleware.service.api.OntologyService;
 import org.springframework.stereotype.Component;
 
+import javax.annotation.Resource;
+
 @Component
 public class CodesSheetGenerator {
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/commons/workbook/generator/CodesSheetGenerator.java:19]: For cleanup, the ff is marked unused:
```
	private GermplasmDataManager germplasmDataManager;
	private ContextUtil contextUtil;
	private WorkbenchDataManager workbenchDataManager;
	private OntologyService ontologyService;
```

File: src/main/java/org/generationcp/commons/workbook/generator/GermplasmAttributesWorkbookExporter.java [hunk 3]
```diff
@@ -1,38 +1,52 @@
...
-	List<UserDefinedField> getSourceItems() {
-		final List<UserDefinedField> attributeTypeColumns = new ArrayList<>();
+	List<Variable> getSourceItems() {
+		final List<Variable> attributeTypeColumns = new ArrayList<>();
 		//columnsInfo is null when exporting germplasm list from Study Manager
 		if (this.columnsInfo != null) {
-			final List<UserDefinedField> attributeTypes = this.germplasmManager.getAllAttributesTypes();
-			final Map<String, UserDefinedField>
-				attributeTypesMap = attributeTypes.stream().collect(Collectors.toMap(u -> u.getFcode().toUpperCase(), u -> u, (u1, u2) -> u1));
+			final VariableFilter variableFilter = new VariableFilter();
```
Comment (darla-leafnode) [inline src/main/java/org/generationcp/commons/workbook/generator/GermplasmAttributesWorkbookExporter.java:34]: There are several places here in Workbench and also in Middleware where we have same code to retrieve passport/attribute variables. Better to centralize in `GermplasmService` or `GermplasmAttributeService` a method:
`List<Variable getAttributeAndPassportVariables(String programUUID)`.

Comment (clarysabel) [reply src/main/java/org/generationcp/commons/workbook/generator/GermplasmAttributesWorkbookExporter.java:34]: I think that this is not really needed, this service will end calling ontologyDataManager to get the information
What I think we should have is a new VariableService in middleware but I will leave out of the scope of this ticket as of now

**Pull request merged**